lib/common/client/middleware/response/raise_error.rb

```diff
@@ -4,9 +4,49 @@ module Client
     module Middleware
       module Response
         class RaiseError < Faraday::Response::Middleware
+          attr_reader :service_acronym, :error_klass, :detail_key, :code_key, :source_key, :meta_key
+
+          def initialize(app, service_acronym, options = {})
+            @service_acronym = service_acronym
+            @error_klass = Common::Exceptions::BackendServiceException
+            @detail_key  = options[:detail_key]  || 'detail'
+            @code_key    = options[:code_key]    || 'code'
+            @source_key  = options[:source_key]  || 'source'
+            @meta_key    = options[:meta_key]    || 'meta'
+
+            super(app)
+          end
+
           def on_complete(env)
             return if env.success?
-            raise Common::Client::Errors::BackendServiceError.new(env.status.to_i, env[:body])
+            raise error_klass.new(i18n_error_key(env), response_values(env))
+          end
+
+          private
+
+          def code(env)
+            "#{service_acronym.upcase}#{env[:body][code_key]}"
+          end
+
+          def i18n_error_key(env)
+            if I18n.exists?("common.exceptions.#{code(env)}")
+              code(env)
+            else
+              Rails.logger.warn "The following exception should be added to locales: \n #{response_values(env)}"
+              'backend_service_exception'
+            end
+          end
+
+          # This method is not to be used for passing options. It is used to notify that
+          # a new error was presented from a backend service that is not currently mapped
+          # in locales
+          def response_values(env)
+            {
+              status: env.status.to_i,
+              detail: env[:body][detail_key],
+              code:   code(env),
+              source: env[:body][source_key]
```

Comment:
I bet this is fine for now, but these lines using `env[:body][{source_key,detail_key,code_key}]``` do make an assumption that the response body contains these things as easily accessible attributes. Is this always the case? I think something was mentioned about some services that return a 200 and then have the error status buried in a JSON response. Might be worth a thought as to how we'd handle that case here. Presumably you'd have to either subclass the middleware or pass it procs that can pull out the keys based on the env.